### PR TITLE
Several fixes for WebGPU

### PR DESCRIPTION
**Pull request by ThiDaGui**
### Context

WebGPU rendering pipeline was not behaving correctly:
-  With firefox nightly (and chrome canary), several WGSL shaders  were not compiling due to type errors.
-  RenderWindow.getStatistics() was calling undefined functions when manipulation WebGPU specialized classes like vtkWebGPURenderWindow or vtkWebGPUPolyDataMapper #2942.

### Results

Now the PBR example run when choosing the webGPU Api.  

### Changes

- Fix various type errors inside WGSL shaders (most of the time, missing floating point for floats)
- Added guards inside RenderWindow.getStatistics() to prevent the call of undefined methods for webGPU specifics objects 

### PR and Code Checklist

- [ ] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [ ] Run `npm run reformat` to have correctly formatted code

### Testing

- [ ] Tested environment:
  - **vtk.js**: 29.4.1
  - **OS**: Linux
  - **Browser**: Firefox nightly 123.0a1

### CODE REVIEW

File: Sources/Rendering/Core/RenderWindow/index.js [hunk 1]
```diff
@@ -97,7 +97,9 @@ function vtkRenderWindow(publicAPI, model) {
   publicAPI.getStatistics = () => {
     const results = { propCount: 0, invisiblePropCount: 0, gpuMemoryMB: 0 };
     model._views.forEach((v) => {
-      results.gpuMemoryMB += v.getGraphicsMemoryInfo() / 1e6;
+      // TODO : remove test when 'vtkWebGPURenderWindow.getGraphicsMemoryInfo()' is implemented
+      if (v.getAllocatedGPUMemoryInBytes)
```
Comment (finetjul) [inline Sources/Rendering/Core/RenderWindow/index.js:101]: shouldn't the test be on `v.getGraphicsMemoryInfo` instead

File: Sources/Rendering/Core/RenderWindow/index.js [hunk 2]
```diff
@@ -97,7 +97,9 @@ function vtkRenderWindow(publicAPI, model) {
   publicAPI.getStatistics = () => {
     const results = { propCount: 0, invisiblePropCount: 0, gpuMemoryMB: 0 };
     model._views.forEach((v) => {
-      results.gpuMemoryMB += v.getGraphicsMemoryInfo() / 1e6;
+      // TODO : remove test when 'vtkWebGPURenderWindow.getGraphicsMemoryInfo()' is implemented
```
Comment (finetjul) [inline Sources/Rendering/Core/RenderWindow/index.js:100]: Not sure this comment is useful. 
Maybe implementing `getGraphicsMemoryInfo()` would result in an asynchronous call hence this whole function would have to be re-written, not just this test.

Comment (natyoufri) [reply Sources/Rendering/Core/RenderWindow/index.js:100]: removed it

File: Sources/Rendering/Core/RenderWindow/index.js (hunk 1 above)
Comment (natyoufri) [reply Sources/Rendering/Core/RenderWindow/index.js]: did this

File: Sources/Rendering/Core/RenderWindow/index.js (hunk 2 above)
Comment (finetjul) [reply Sources/Rendering/Core/RenderWindow/index.js:100]: there is still one comment remaining.

Comment (natyoufri) [reply Sources/Rendering/Core/RenderWindow/index.js]: oups, bad ctrl-z at the wrong time

**Comment by finetjul** [comment]
@ThiDaGui @ThiDaGui2 @natyoufri can you please squash your PR into 1 or 2 commits ?

**Comment by natyoufri** [comment]
Yes. i will do it tomorrow morning. sorry for the delay

**Pull request merged**